### PR TITLE
Feature/edit with nominatim

### PR DESCRIPTION
**Pull request by Redd87**
As my previous PR adding the event edit feature was developed concurrently with @Zafouche 's PR that added the nominatim location picker in the EventCreation composable, the location picker wasn't added to the Edit screen. This PR aims to fix this. It extracts the location picker in the EventEditComponents file and changes EventCreation.kt accordingly, and adds it to EventEdit. I also added some documentation to EventEditComponents as it was lacking. 

This is currently a draft PR as further tests are needed in EventEdit.

### CODE REVIEW

File: app/src/main/java/com/android/unio/ui/components/EventEditComponents.kt
Comment (Romainhir) [inline app/src/main/java/com/android/unio/ui/components/EventEditComponents.kt:143]: Empty onClick parameter. Is there really nothing to do on click here (not even a Toast or something) ?

**Review by Romainhir — CHANGES_REQUESTED** [review]
Nice job ! I only have one question otherwise, LGTM.

File: app/src/main/java/com/android/unio/ui/components/EventEditComponents.kt
Comment (Zafouche) [reply app/src/main/java/com/android/unio/ui/components/EventEditComponents.kt:143]: Yeah this comes from the implementation that was done in my PR of the `Location` input in the `Event` creation screen. which he simply refactored but did not modify it in anyway (or at least I don't see anything that has been modified).

@Redd87 I suggest you add a `Toast` telling the user to continue writing the address, such that the one that he wants shows up in the first three suggestions. 

Or you can make it such that it adds more suggestions to the `Dropdown`, but I don't think allowing the user to extend the Dropdown infinitely is a good idea. 

We could refine it later on by allowing to add a couple of suggestions, only once, but this is not important for this PR.

The first solution, IMO, is the best, which simply consists in adding a `Toast`. 
Please add it if possible, otherwise LGTM as well.

Comment (Romainhir) [reply app/src/main/java/com/android/unio/ui/components/EventEditComponents.kt:143]: Ok, I'll approve for now then. Thank you for the answer

Comment (Zafouche) [reply app/src/main/java/com/android/unio/ui/components/EventEditComponents.kt:143]: Still it'd be great if @Redd87 could add the `Toast` before merging.

Comment (Redd87) [reply app/src/main/java/com/android/unio/ui/components/EventEditComponents.kt:143]: @Zafouche ok I'll let you add the toast

**Pull request merged**
